sources/lib/api/gcp/bigquery/_udf.py

```diff
@@ -89,50 +73,13 @@ def _build_js(inputs, outputs, name, implementation):
     output_fields = json.dumps(output_fields, sort_keys=True)
 
     # Build the JS from the individual bits with proper escaping of the implementation
-    return '%s=%s;\nbigquery.defineFunction(\'%s\', %s, %s, %s);' %\
-           (name, implementation.replace('"', '\\"'), name, input_fields, output_fields, name)
-
-
-class FunctionEvaluation(object):
-
-  def __init__(self, implementation, data):
-    self._implementation = implementation
-    self._data = data
-
-  @property
-  def data(self):
-    return self._data
-
-  @property
-  def implementation(self):
-    return self._implementation
-
-
-class UDF(object):
-  """Represents a BigQuery UDF declaration.
-  """
-
-  def __init__(self, inputs, outputs, name, implementation):
-    """Initializes a Function object from its pieces.
-
-    Args:
-      inputs: a list of string field names representing the schema of input.
-      outputs: a list of name/type tuples representing the schema of the output.
-      name: the name of the javascript function
-      implementation: a javascript function implementing the logic.
-    Raises:
-      Exception if the name is invalid.
-      """
-    self._inputs = inputs
-    self._outputs = outputs
-    self._name = name
-    self._implementation = implementation
-
-  def __call__(self, data):
-    if issubclass(type(data), list):
-      return FunctionEvaluation(self._implementation, data)
-    else:
-      return FunctionCall(data, self._inputs, self._outputs, self._name, self._implementation)
+    if support_code is None:
+      support_code = ''
+    return ('{code}\n{name}={implementation};\n' + \
+            'bigquery.defineFunction(\'{name}\', {inputs}, {outputs}, {name});')\
+                .format(code=support_code,
+                        name=name,
+                        implementation=implementation.replace('"', '\\"'),
```

Comment:
Do we also need to escape the escapes ... so \ -> \\ before escaping quotes of course.